rosbag2_transport/src/rosbag2_transport/play_options_from_node_params.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 Patrick Roncagliolo.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "rosbag2_transport/qos.hpp"
+#include "rosbag2_transport/play_options.hpp"
+#include "rosbag2_transport/utils/param_utils.hpp"
+#include "rosbag2_transport/play_options_from_node_params.hpp"
+
+namespace rosbag2_transport
+{
+
+PlayOptions get_play_options_from_node_params(std::shared_ptr<rclcpp::Node> node)
+{
+  PlayOptions play_options{};
+  auto desc_raqs = param_utils::int_param_description(
+    "Read ahead queue size (messages)",
+    1,
+    std::numeric_limits<int64_t>::max());
+  auto read_ahead_queue_size_ = node->declare_parameter<int64_t>(
+    "read_ahead_queue_size",
+    1000,
+    desc_raqs);
+  play_options.read_ahead_queue_size = static_cast<uint64_t>(read_ahead_queue_size_);
+
+  play_options.node_prefix = node->declare_parameter<std::string>(
+    "node_prefix",
+    "");
+
+  auto desc_rate = param_utils::float_param_description(
+    "Playback rate (hz)",
+    0.000001,
+    std::numeric_limits<float>::max());
+  play_options.rate = node->declare_parameter<float>(
+    "rate",
+    1.0,
+    desc_rate);
+
+  play_options.topics_to_filter = node->declare_parameter<std::vector<std::string>>(
+    "topics_to_filter",
+    std::vector<std::string>());
+
+  play_options.topics_regex_to_filter = node->declare_parameter<std::string>(
+    "topics_regex_to_filter",
+    "");
+
+  play_options.topics_regex_to_exclude = node->declare_parameter<std::string>(
+    "topics_regex_to_exclude",
+    "");
+
+  // TODO(roncapat)
+  // std::unordered_map<std::string, rclcpp::QoS> topic_qos_profile_overrides = {};
```

Comment:
The same comment as for the `recorder_options`. We can parse qos profiles from file.